pkg/kubelet/cm/topologymanager/topology_manager_test.go

```diff
@@ -104,22 +104,18 @@ func TestGetAffinity(t *testing.T) {
 	}
 }
 
-func TestCalculateAffinity(t *testing.T) {
```

Comment:
It's hard to verify taht all the same tests are still here after this change. Can you walk me through your changes a bit. It looks like there are some additions, some removals, and some outright changes to the tests.

Ideally this would have been split into 3 separate commits to make it easier to review (one for changes, one for additions, and one for removals). But if you can walk me through the changes a bit more that should be OK:

Reply:
So 90% of the tests where moved to the policies.
the main change is that the merge method now accepts : `[]map[string][]TopologyHint`  
so the test case input needed to be changed slightly as it no longer deals with hint providers directly.

in topology manager we test that it is able to properly coalesce the info from hint providers and pass it to the policy.Merge method.

The tests surrounding merge where moved to `policy_best_effort_test.go:testOptimalMerge`
with the needed change of translating hint providers  `GetTopologyHints` output to `[]map[string][]TopologyHint`

The test cases that where removed is because they where not valid for the new Mege method.
however what those tests were testing was kept in `TestCalculateAffinity` (manly around edge cases of what a hint provider may return in `GetTopologyHints`)